src/mtype.d

```diff
@@ -682,7 +682,7 @@ public:
                 {
                     goto Ldistinct;
                 }
-                const(StorageClass) sc = STCref | STCin | STCout | STClazy;
+                const(StorageClass) sc = STCref | STCin | STCout | STClazy | STCconst | STCimmutable | STCshared;
```

Comment:
Maybe `Type::covariant` is not the best way to check if two `TypeFunction`'s match.